tests/config_file_test.py

```diff
@@ -39,128 +39,109 @@ def assert_eq(self, result, expected, error_message):
-                        }
-                }
+        def assert_sub_value_eq(exp_val, res_val):
+            if isinstance(exp_val, dict):
+                assert_sub_dict_eq(exp_val, res_val)
+            elif isinstance(exp_val, list):
+                assert_sub_list_eq(exp_val, res_val)
+            else:
+                self.assertEqual(exp_val, res_val)
+        
+        def assert_sub_list_eq(exp, res):
+            '''
+            compares the result list against the expected list recursively, and
+            raises an assertion error if ti's values or sub-values don't match
+            '''
+            self.assertIsInstance(exp, list, 'expected is not dict')
+            self.assertIsInstance(res, list, 'result is not list')
             
-            yml = ConfigFileLoader.load_root_config('config-test/user-sync-config-test.yml')
+            self.assertEqual(len(exp), len(res), 'expected and result lists don\'t have the same number of entries')
             
-            # test path updating
-            self.assert_eq(yml['dashboard']['owning'], os.path.abspath('config-test/dashboard/dashboard-owning-config-test.yml'), 'owning dashboard configuration path is incorrect')
-            self.assert_eq(yml['dashboard']['accessor_config_filename_format'], os.path.abspath('config-test/accessors-test/dashboard-accessor-{organization_name}-config.yml'), 'accessor dashboard config file format is incorrect')
-            self.assert_eq(yml['dashboard']['accessors']['test-org-1']['enterprise']['priv_key_path'], os.path.abspath('keys/test-org-1/private-key-test.key'), 'accessor test_org_1 dashboard config private key path is incorrect')
-            self.assert_eq(yml['dashboard']['accessors']['test-org-2'], os.path.abspath('config-test/accessors/xyz/accessor-config-test.yml'), 'accessor test_org_2 dashboard config file path is incorrect')
-            self.assert_eq(yml['logging']['file_log_directory'], os.path.abspath('config-test/logs'), 'logging path is incorrect')
+            for exp_item, res_item in zip(exp, res):
+                assert_sub_value_eq(exp_item, res_item)
+    
+        def assert_sub_dict_eq(exp, res):
+            '''
+            compares the result dict against the expected dict recursively, and
+            raises an assertion error if it's values or sub-values don't match.
+            '''
+            self.assertIsInstance(exp, dict, 'expected is not dict')
+            self.assertIsInstance(res, dict, 'result is not dict')
+            
+            self.assertEqual(len(exp.keys()), len(res.keys()), 'expected and result dicts don\'t have the same number of keys')
+    
+            for key in exp.keys():
+                assert_sub_value_eq(exp[key], res[key])
+            
+        try:
+            assert_sub_dict_eq(exp, res)
+        except AssertionError as e:
+            raise AssertionError('%s\n%s' % (err_msg, e.message))
+        
+    def test_load_root_config(self):
```

Comment:
It would be nice to have just one function that does the test against any pair of files, or even a pair of "file groups" (a set of related config files - master and sub, and a set of "correct answers") where not all the files in a group would need to be in the same directory.  I think, rather than explicitly specifying the paths to convert in the function itself, you could structure the "correct answer" file so that  it had a marker in its keys (or its values), for example, file: as a prefix on the key or value, and the presence of that marker means the key/value holds a relativized form.  Then the test function would use those markers to find the keys/values in the answer, convert them to standard form, and then the answer file would be directly compared against the source file.  (Basically, you would take the source file and copy it, then mark some of its keys and fix the values of those keys, and that would be the answer file.)